File: backend/app/schemas/document.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime


DOCUMENT_CATEGORIES = {"opening", "closing", "ups", "events", "business"}


class DocumentCreate(BaseModel):
    title: str
    category: str
    description: Optional[str] = None
    is_pinned: Optional[bool] = False


class DocumentUpdate(BaseModel):
    title: Optional[str] = None
    category: Optional[str] = None
    description: Optional[str] = None
    is_pinned: Optional[bool] = None


class DocumentOut(BaseModel):
    id: int
    title: str
    category: str
    description: Optional[str]
    original_filename: str
    mime_type: str
    file_size: int
    is_pinned: bool
    uploaded_by_user_id: Optional[int]
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/document.ts
export type DocumentCategory = "opening" | "closing" | "ups" | "events" | "business";

export interface Document {
  id: number;
  title: string;
  category: DocumentCategory;
  description?: string | null;
  original_filename: string;
  mime_type: string;
  file_size: number;
  is_pinned: boolean;
  uploaded_by_user_id?: number | null;
  created_at: string;
  updated_at: string;
}

export interface DocumentUpdate {
  title?: string;
  category?: DocumentCategory;
  description?: string | null;
  is_pinned?: boolean;
}
